test/e2e/replica_set_custom_persistent_volume/replica_set_custom_persistent_volume.go

```diff
@@ -0,0 +1,141 @@
+)
+
+func TestMain(m *testing.M) {
+	code, err := e2eutil.RunTest(m)
+	if err != nil {
+		fmt.Println(err)
+	}
+	os.Exit(code)
+}
+
+func getPersistentVolumeLocal(name string, localPath string, label string) corev1.PersistentVolume {
+	return corev1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: e2eutil.OperatorNamespace,
+			Labels:    map[string]string{"type": label},
+		},
+		Spec: corev1.PersistentVolumeSpec{
+			PersistentVolumeSource: corev1.PersistentVolumeSource{
+				Local: &corev1.LocalVolumeSource{
+					Path: localPath,
+				},
+			},
+			AccessModes:      []corev1.PersistentVolumeAccessMode{"ReadWriteOnce"},
+			StorageClassName: "default",
+			NodeAffinity: &corev1.VolumeNodeAffinity{
+				Required: &corev1.NodeSelector{
+					NodeSelectorTerms: []corev1.NodeSelectorTerm{
+						{
+							MatchExpressions: []corev1.NodeSelectorRequirement{
+								{
+									Key:      "kubernetes.io/hostname",
+									Operator: "In",
+									Values:   []string{"kind-control-plane"},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getVolumes(ctx *e2eutil.Context, volumeType string, members int) []corev1.PersistentVolume {
+	volumes := make([]corev1.PersistentVolume, members)
+	for i := 0; i < members; i++ {
+		volumes[i] = getPersistentVolumeLocal(
+			fmt.Sprintf("%s-volume-%d", volumeType, i),
+			fmt.Sprintf("/opt/data/mongo-%s-%d", volumeType, i),
+			volumeType,
+		)
+	}
+	return volumes
+}
+
+func getPvc(pvcType string, mdb v1.MongoDBCommunity) corev1.PersistentVolumeClaim {
+	name := ""
+	if pvcType == "logs" {
+		name = mdb.LogsVolumeName()
+	} else {
+		name = mdb.DataVolumeName()
+	}
+	defaultString := "default"
+	return corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
+			AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"type": pvcType},
+			},
+			Resources: corev1.ResourceRequirements{
+				Requests: map[corev1.ResourceName]resource.Quantity{"storage": *resource.NewScaledQuantity(int64(8), resource.Giga)},
+			},
+			StorageClassName: &defaultString,
+		},
+	}
+}
+
+func TestReplicaSetCustomPersistentVolumes(t *testing.T) {
+	ctx := setup.Setup(t)
+	defer ctx.Teardown()
+
+	mdb, user := e2eutil.NewTestMongoDB(ctx, "mdb0", "")
+	mdb.Spec.StatefulSetConfiguration.SpecWrapper.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{
+		getPvc("data", mdb),
+		getPvc("logs", mdb),
+	}
+	volumesToCreate := getVolumes(ctx, "data", mdb.Spec.Members)
+	volumesToCreate = append(volumesToCreate, getVolumes(ctx, "logs", mdb.Spec.Members)...)
+
+	for i := range volumesToCreate {
+		err := e2eutil.TestClient.Create(context.TODO(), &volumesToCreate[i], &e2eutil.CleanupOptions{TestContext: ctx})
+		assert.NoError(t, err)
+	}
+	scramUser := mdb.GetScramUsers()[0]
```

Comment:
[q] isn't `user` which is already in scope the same as `scramUser`?

Reply:
I copied the skeleton for this test from another one, but let me double check!

Reply:
they have different types

`user` is `MongoDBUser`
`scramUser` is `scram.User`

I guess we could avoid duplicating and just converting one to the other, but this is basically what `GetScramUsers()` does, I think it's ok to leave it like this, what do you think?